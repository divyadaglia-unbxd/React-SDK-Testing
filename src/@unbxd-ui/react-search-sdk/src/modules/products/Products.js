import React from 'react';
import PropTypes from 'prop-types';

import ProductsContainer from './ProductsContainer';
import { AppContextConsumer } from '../../common/context';
import { hasUnbxdSearchWrapperContext } from '../../common/utils';

/**
 * Component to manage products.
 * Products also manages the pagination options for the search results here.
 */
const Products = (props) => {
    return (
        <AppContextConsumer>
            {(appState) => {
                if (appState === undefined) {
                    hasUnbxdSearchWrapperContext(Products.displayName);
                }

                const {
                    unbxdCore,
                    unbxdCoreStatus,
                    helpers,
                    unbxdState,
                    priceUnit
                } = appState;
                const { viewType, pageSize, sort } = unbxdState;

                return (
                    <ProductsContainer
                        unbxdCore={unbxdCore}
                        unbxdCoreStatus={unbxdCoreStatus}
                        helpers={helpers}
                        viewType={viewType}
                        pageSize={pageSize}
                        sort={sort}
                        priceUnit={priceUnit}
                        {...props}
                    />
                );
            }}
        </AppContextConsumer>
    );
};

Products.displayName = 'Products';

Products.defaultProps = {
    perRow: 4,
    pageSize: 10,
    variantsCount: 0,
    paginationType: 'FIXED_PAGINATION',
    heightDiffToTriggerNextPage: 300,
    showLoader: false,
    productIdAttribute: 'uniqueId',
    groupBy: '',
    variantAttributesMap: {},
    onResultsLoaded:()=>{}
};

Products.propTypes = {
    /**
     * Number of products to be shown per row.
     */
    perRow: PropTypes.number,
    /**
     * Required pagination type. Possible options are `INFINITE_SCROLL`, `CLICK_N_SCROLL` and `FIXED_PAGINATION`.
     */
    paginationType: PropTypes.string,
    /**
     * Height difference to trigger for next page in case of paginationType `INFINITE_SCROLL`.
     */
    heightDiffToTriggerNextPage: PropTypes.number,
    /**
     * Custom load more component instance
     */
    loadMoreComponent: PropTypes.element,
    /**
     * Unique attribute of the product
     */
    productIdAttribute: PropTypes.string,
    /**
     * Mapping of catalog Product fields to SDK Product fields.
     */
    attributesMap: PropTypes.object.isRequired,
    /**
     * Show variants to a product.
     */
    showVariants: PropTypes.bool,
    /**
     * Number of variants to fetch.
     */
    variantsCount: PropTypes.number,
    /**
     * Mapping of catalog Product variant fields to SDK Product variant fields.
     */
    variantAttributesMap: PropTypes.object,
    /**
     * Should loader be shown
     */
    showLoader: PropTypes.bool,
    /**
     * Custom loader component instance
     */
    loaderComponent: PropTypes.element,
    /**
     * Callback function triggered on click of a product.
     */
    onProductClick: PropTypes.func,
    /**
     * Callback function triggered on zero results.
     */
    onZeroResults: PropTypes.func,
    /**
     * Display swatches
     */
    showSwatches: PropTypes.bool,
    /**
     * Group variants by the attribute
     */
    groupBy: PropTypes.string,
    /**
     * Swatch attributes that change on click on of the swatch
     */
    swatchAttributesMap: PropTypes.object,
    /**
     * Custom swatch component instance
     */
    swatchItemComponent: PropTypes.element,
    /**
     * Custom product item component instance
     */
    productItemComponent: PropTypes.element,
    /**
     *  Custom zero result component instance
     */
    zeroResultsComponent: PropTypes.element,
    /**
     *  callback on each new results load
     */
     onResultsLoaded: PropTypes.func
};

export default Products;
