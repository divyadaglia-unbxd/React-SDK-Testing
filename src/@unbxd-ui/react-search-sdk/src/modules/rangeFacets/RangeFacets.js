import React from 'react';
import PropTypes from 'prop-types';

import { AppContextConsumer } from '../../common/context';
import { hasUnbxdSearchWrapperContext } from '../../common/utils';
import RangeFacetsContainer from './RangeFacetsContainer';

/**
 * Component to manage range facets.
 */
const RangeFacets = (props) => {
    return (
        <AppContextConsumer>
            {(appState) => {
                if (appState === undefined) {
                    hasUnbxdSearchWrapperContext(RangeFacets.displayName);
                }

                const {
                    unbxdCore,
                    unbxdCoreStatus,
                    helpers,
                    unbxdState,
                    priceUnit
                } = appState;
                const { enableApplyFilters, selectedRangeFacets } = unbxdState;

                return (
                    <RangeFacetsContainer
                        unbxdCore={unbxdCore}
                        unbxdCoreStatus={unbxdCoreStatus}
                        helpers={helpers}
                        enableApplyFilters={enableApplyFilters}
                        selectedRangeFacets={selectedRangeFacets}
                        priceUnit={priceUnit}
                        {...props}
                    />
                );
            }}
        </AppContextConsumer>
    );
};

RangeFacets.displayName = 'RangeFacets';

RangeFacets.defaultProps = {
    collapsible: true,
    minViewMore: 3,
    enableViewMore: false,
    applyMultiple: false
};

RangeFacets.propTypes = {
    /**
     * Collapse facet values
     */
    collapsible: PropTypes.bool,
    /**
     * Bool value to enable/disable viewMore.
     */
    enableViewMore: PropTypes.bool,
    /**
     * Min value for viewMore to be enabled.
     */
    minViewMore: PropTypes.number,
    /**
     * Apply multiple filters on the same facet
     */
    applyMultiple: PropTypes.bool,
    /**
     * Custom facet item component instance.
     */
    facetItemComponent: PropTypes.element,
    /**
     * Label for the module.
     */
    label: PropTypes.node,
    /**
     * Callback for facet change.
     */
    onFacetClick: PropTypes.func,
    /**
     * Callback to format the facets.
     */
    transform: PropTypes.func
};

export default RangeFacets;
