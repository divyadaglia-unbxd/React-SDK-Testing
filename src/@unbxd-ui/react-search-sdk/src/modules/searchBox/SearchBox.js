import React from 'react';
import PropTypes from 'prop-types';

import { AppContextConsumer } from '../../common/context';
import { hasUnbxdSearchWrapperContext } from '../../common/utils';
import SearchBoxContainer from './SearchBoxContainer';

/**
 * Component to display merchandising banners.
 */
const SearchBox = (props) => {
    return (
        <AppContextConsumer>
            {(appState) => {
                if (appState === undefined) {
                    hasUnbxdSearchWrapperContext(SearchBox.displayName);
                }

                const {
                    unbxdCore,
                    unbxdCoreStatus,
                    helpers,
                    productType,
                    unbxdState
                } = appState;

                const { query } = unbxdState;

                return (
                    <SearchBoxContainer
                        unbxdCore={unbxdCore}
                        unbxdCoreStatus={unbxdCoreStatus}
                        helpers={helpers}
                        productType={productType}
                        query={query}
                        {...props}
                    />
                );
            }}
        </AppContextConsumer>
    );
};

SearchBox.displayName = 'SearchBox';

SearchBox.defaultProps = {
    autoFocus: false,
    clearable: false,
    showLoader: false
};

SearchBox.propTypes = {
    /**
     * Should the searchbox be focused by default.
     */
    autoFocus: PropTypes.bool,
    /**
     * Should the searchbox be clearable.
     */
    clearable: PropTypes.bool,
    /**
     * Hook for search query. The function should return `true` if the search is to be triggered, false otherwise.
     */
    onSubmit: PropTypes.func,
    /**
     * Hook for clearing the search query. The function should return `true` if the searchbox is to be cleared, false otherwise.
     */
    onClear: PropTypes.func,
    /**
     * Should loader be shown
     */
    showLoader: PropTypes.bool,
    /**
     * Custom input component instance
     */
    inputComponent: PropTypes.element,
    /**
     * Custom submit component instance
     */
    submitComponent: PropTypes.element,
    /**
     * Custom reset component instance
     */
    clearComponent: PropTypes.element,
    /**
     * Default search query
     */
    defaultSearch: PropTypes.string,
    /**
     * SearchBox placeholder
     */
    placeholder: PropTypes.string
};

export default SearchBox;
