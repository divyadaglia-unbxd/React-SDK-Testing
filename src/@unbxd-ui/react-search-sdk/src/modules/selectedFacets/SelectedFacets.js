import React from 'react';
import PropTypes from 'prop-types';

import { AppContextConsumer } from '../../common/context';
import { hasUnbxdSearchWrapperContext } from '../../common/utils';
import SelectedFacetsContainer from './SelectedFacetsContainer';

/**
 * Component to manage selected facets.
 */
const SelectedFacets = (props) => {
    return (
        <AppContextConsumer>
            {(appState) => {
                if (appState === undefined) {
                    hasUnbxdSearchWrapperContext(SelectedFacets.displayName);
                }

                const {
                    unbxdCore,
                    unbxdCoreStatus,
                    unbxdState,
                    priceUnit,
                    helpers: { getUpdatedResults },
                    productType,
                    helpers
                } = appState;

                const { applyMultiple } = unbxdState;

                return (
                    <SelectedFacetsContainer
                        unbxdCore={unbxdCore}
                        unbxdCoreStatus={unbxdCoreStatus}
                        helpers={helpers}
                        priceUnit={priceUnit}
                        getUpdatedResults={getUpdatedResults}
                        applyMultiple={applyMultiple}
                        productType={productType}
                        {...props}
                    />
                );
            }}
        </AppContextConsumer>
    );
};

SelectedFacets.displayName = 'SelectedFacets';

SelectedFacets.propTypes = {
    /**
     * Custom facet item component instance.
     */
    facetItemComponent: PropTypes.element,
    /**
     * Label for the module.
     */
    label: PropTypes.node
};

export default SelectedFacets;
