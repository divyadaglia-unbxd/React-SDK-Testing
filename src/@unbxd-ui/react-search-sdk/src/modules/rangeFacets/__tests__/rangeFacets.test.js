import renderer from 'react-test-renderer';
import { render, waitFor, fireEvent } from '@testing-library/react';
import React from 'react';
import Products from '../../products/index';
import RangeFacets from '../index';
import UnbxdSearchWrapper from '../../../UnbxdSearchWrapper';
import SearchBox from '../../searchBox';
import { searchResponse } from './mocks/searchMock';
import { facetResponse } from './mocks/searchMock';
import SelectedFacets from '../../selectedFacets/index';
import FacetActions from '../../facetActions/index';

export const FacetItemComponent = ({ itemData, onClick, priceUnit }) => {
    const { from, end, facetName, isSelected = false } = itemData;
    const { name: fromName, count, dataId: fromDataId } = from;
    const { name: ToName, dataId: toDataId } = end;

    const handleClick = () => {
        onClick(itemData);
    };

    return (
        <div
            key={`${facetName}_${fromDataId}-${toDataId}`}
            className={`UNX-facet__item test-class ${
                isSelected ? '-selected' : ''
            }`}
            onClick={handleClick}
        >
            <div className="-checkbox" />
            <div className="-label">
                {priceUnit}
                {fromName} - {priceUnit}
                {ToName}
            </div>
            <div className="-count">({count})</div>
        </div>
    );
};

const FacetItemComponentSelected = ({ itemData, onClick, priceUnit }) => {
    const { name, type, dataId } = itemData;
    const handleClick = () => {
        onClick(itemData);
    };

    let selectedFacetMarkup = null;
    if (type === 'TEXT_FACET') {
        selectedFacetMarkup = <span className="text-selected">{name}</span>;
    }
    if (type === 'RANGE_FACET') {
        const [valMin, valMax] = dataId.split(' TO ');
        selectedFacetMarkup = (
            <span className="range-selected">
                {priceUnit} {valMin} - {priceUnit} {valMax}
            </span>
        );
    }
    if (type === 'MULTILEVEL_FACET') {
        selectedFacetMarkup = <span>{name}</span>;
    }

    return (
        <div className="UNX-selectedFacets__item" onClick={handleClick}>
            {selectedFacetMarkup} <span className="-cross" />
        </div>
    );
};

// establish API mocking before all tests
beforeAll(() => {
    window.fetch = jest.fn((request) => {
        if (request.includes('filter=price:[200 TO 300]')) {
            return Promise.resolve({
                json: () => Promise.resolve(facetResponse)
            });
        }
        return Promise.resolve({
            json: () => Promise.resolve(searchResponse)
        });
    });
});

const attributesMap = {
    title: 'title',
    uniqueId: 'uniqueId',
    imageUrl: 'imageUrl',
    price: 'RRP_Price',
    productUrl: 'productUrl'
};

test('Match Snapshot for range facets', async () => {
    const tree = renderer.create(
        <UnbxdSearchWrapper
            siteKey="wildearthclone-neto-com-au808941566310465"
            apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
        >
            <SelectedFacets />
            <RangeFacets />
            <Products attributesMap={attributesMap} />
            <div>
                <SearchBox defaultSearch="shoes" />
            </div>
        </UnbxdSearchWrapper>
    );
    await waitFor(() => tree.toJSON());
    expect(tree.toJSON()).toMatchSnapshot();
});

test('Test range facet click', async () => {
    const { getByText } = render(
        <>
            <UnbxdSearchWrapper
                siteKey="wildearthclone-neto-com-au808941566310465"
                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
            >
                <RangeFacets />
                <Products attributesMap={attributesMap} />
                <div>
                    <SearchBox defaultSearch="shoes" />
                </div>
            </UnbxdSearchWrapper>
        </>
    );

    await waitFor(async () => {
        expect(getByText('$ 200 - $ 300 - 60')).toBeInTheDocument();
        fireEvent.click(getByText('$ 200 - $ 300 - 60'));
    });
    await waitFor(() => {
        expect(
            getByText(
                'Salomon OUTblast TS CSWP Mens Hiking Boots - Black/Black/Black'
            )
        ).toBeInTheDocument();
    });
});

test('Test range facet with FacetItemComponent', async () => {
    const { container } = render(
        <>
            <UnbxdSearchWrapper
                siteKey="wildearthclone-neto-com-au808941566310465"
                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
            >
                <RangeFacets facetItemComponent={<FacetItemComponent />} />
                <Products attributesMap={attributesMap} />
                <div>
                    <SearchBox defaultSearch="shoes" />
                </div>
            </UnbxdSearchWrapper>
        </>
    );

    await waitFor(() => {
        expect(container.getElementsByClassName('test-class').length).toBe(9);
    });
});

test('Test range facet click with FacetItemComponent', async () => {
    const { getByText } = render(
        <>
            <UnbxdSearchWrapper
                siteKey="wildearthclone-neto-com-au808941566310465"
                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
            >
                <RangeFacets facetItemComponent={<FacetItemComponent />} />
                <Products attributesMap={attributesMap} />
                <div>
                    <SearchBox defaultSearch="shoes" />
                </div>
            </UnbxdSearchWrapper>
        </>
    );

    await waitFor(async () => {
        expect(getByText('$200 - $300')).toBeInTheDocument();
        fireEvent.click(getByText('$200 - $300'));
    });
    await waitFor(() => {
        expect(
            getByText(
                'Salomon OUTblast TS CSWP Mens Hiking Boots - Black/Black/Black'
            )
        ).toBeInTheDocument();
    });
});
test('Test selected facet click on range Facet', async () => {
    const { getByText, container } = render(
        <>
            <UnbxdSearchWrapper
                siteKey="wildearthclone-neto-com-au808941566310465"
                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
            >
                <FacetActions showApplyFilter={false} showClearFilter={false} />
                <SelectedFacets />
                <RangeFacets />
                <Products attributesMap={attributesMap} />
                <div>
                    <SearchBox defaultSearch="shoes" />
                </div>
            </UnbxdSearchWrapper>
        </>
    );

    await waitFor(async () => {
        const aRange = getByText('$ 200 - $ 300 - 60');
        expect(aRange).toBeInTheDocument();
        fireEvent.click(aRange);
        
    });
    await waitFor(() => {
        expect(
            container.getElementsByClassName('UNX-facet__item -selected')
                .length
        ).toBe(1);
    });
});

test('Test selected facet click on range Facet with FacetItemComponent', async () => {
    const { getByText, container } = render(
        <>
            <UnbxdSearchWrapper
                siteKey="wildearthclone-neto-com-au808941566310465"
                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
            >
                <FacetActions showApplyFilter={false} showClearFilter={false} />
                <SelectedFacets
                    facetItemComponent={<FacetItemComponentSelected />}
                />
                <RangeFacets />
                <Products attributesMap={attributesMap} />
                <div>
                    <SearchBox defaultSearch="shoes" />
                </div>
            </UnbxdSearchWrapper>
        </>
    );

    await waitFor(async () => {
        expect(getByText('$ 200 - $ 300 - 60')).toBeInTheDocument();
        fireEvent.click(getByText('$ 200 - $ 300 - 60'));
    });
    await waitFor(() => {
        expect(container.getElementsByClassName('-selected').length).toBe(
            1
        );
    });
});
