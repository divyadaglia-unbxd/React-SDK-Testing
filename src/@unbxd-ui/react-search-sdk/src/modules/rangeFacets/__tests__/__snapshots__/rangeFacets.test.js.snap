// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match Snapshot for range facets 1`] = `
Array [
  <div
    className="UNX-rangefacet__container"
  >
    <div
      className="UNX-facet__element open"
    >
      <div
        className="UNX-facet__headerContainer"
      >
        <div
          className="UNX-facet__header"
        >
          FILTER BY PRICE
          <span
            aria-label="collapse facets"
            className="-collapse-icon"
            data-unx_name="price"
            onClick={[Function]}
            role="button"
            tabIndex={0}
          />
        </div>
      </div>
      <div
        className="UNX-facet__list "
      >
        <div
          className="UNX-facet__item "
          data-facet-name="price"
          data-id="[0-100]"
          data-testid="UNX_rangeFacet__facetItem"
          onClick={[Function]}
          role="button"
          tabIndex={0}
        >
          $
           
          0
           - 
          $
           
          100
           - 
          21
        </div>
        <div
          className="UNX-facet__item "
          data-facet-name="price"
          data-id="[100-200]"
          data-testid="UNX_rangeFacet__facetItem"
          onClick={[Function]}
          role="button"
          tabIndex={0}
        >
          $
           
          100
           - 
          $
           
          200
           - 
          18
        </div>
        <div
          className="UNX-facet__item "
          data-facet-name="price"
          data-id="[200-300]"
          data-testid="UNX_rangeFacet__facetItem"
          onClick={[Function]}
          role="button"
          tabIndex={0}
        >
          $
           
          200
           - 
          $
           
          300
           - 
          60
        </div>
        <div
          className="UNX-facet__item "
          data-facet-name="price"
          data-id="[300-400]"
          data-testid="UNX_rangeFacet__facetItem"
          onClick={[Function]}
          role="button"
          tabIndex={0}
        >
          $
           
          300
           - 
          $
           
          400
           - 
          38
        </div>
        <div
          className="UNX-facet__item "
          data-facet-name="price"
          data-id="[400-600]"
          data-testid="UNX_rangeFacet__facetItem"
          onClick={[Function]}
          role="button"
          tabIndex={0}
        >
          $
           
          400
           - 
          $
           
          600
           - 
          13
        </div>
        <div
          className="UNX-facet__item "
          data-facet-name="price"
          data-id="[600-700]"
          data-testid="UNX_rangeFacet__facetItem"
          onClick={[Function]}
          role="button"
          tabIndex={0}
        >
          $
           
          600
           - 
          $
           
          700
           - 
          1
        </div>
        <div
          className="UNX-facet__item "
          data-facet-name="price"
          data-id="[700-900]"
          data-testid="UNX_rangeFacet__facetItem"
          onClick={[Function]}
          role="button"
          tabIndex={0}
        >
          $
           
          700
           - 
          $
           
          900
           - 
          4
        </div>
        <div
          className="UNX-facet__item "
          data-facet-name="price"
          data-id="[900-1000]"
          data-testid="UNX_rangeFacet__facetItem"
          onClick={[Function]}
          role="button"
          tabIndex={0}
        >
          $
           
          900
           - 
          $
           
          1000
           - 
          3
        </div>
        <div
          className="UNX-facet__item "
          data-facet-name="price"
          data-id="[1000-1100]"
          data-testid="UNX_rangeFacet__facetItem"
          onClick={[Function]}
          role="button"
          tabIndex={0}
        >
          $
           
          1000
           - 
          $
           
          1100
           - 
          1
        </div>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="UNX-products__container"
    >
      <div
        className="UNX-products__list -grid grid-cols-4"
      >
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/vasque-breeze-all-terrain-gtx-mens-hiking-boots-br/V7040W"
            onClick={[Function]}
          >
            <img
              alt="Vasque Breeze All-Terrain GTX Mens Hiking Boots - Brown Olive/Bossa Nova"
              className="-image"
              src="https://www.wildearth.com.au/assets/full/V7040W.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Vasque Breeze All-Terrain GTX Mens Hiking Boots - Brown Olive/Bossa Nova
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                369.99
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/vasque-breeze-all-terrain-gtx-womens-hiking-boots/V7023W"
            onClick={[Function]}
          >
            <img
              alt="Vasque Breeze All-Terrain GTX Womens Hiking Boots - Gargyle"
              className="-image"
              src="https://www.wildearth.com.au/assets/full/V7023W.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Vasque Breeze All-Terrain GTX Womens Hiking Boots - Gargyle
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                369.99
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/scarpa-phantom-6000-mountaineering-boots-black-ora/SCA40017-BlackOrange"
            onClick={[Function]}
          >
            <img
              alt="Scarpa Phantom 6000 Unisex Waterproof Mountaineering Boots"
              className="-image"
              src="https://www.wildearth.com.au/assets/full/SCA40017-BlackOrange.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Scarpa Phantom 6000 Unisex Waterproof Mountaineering Boots
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                1099.95
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/scarpa-mont-blanc-pro-gtx-goretex-mountaineering-b/SCA40013"
            onClick={[Function]}
          >
            <img
              alt="Scarpa Mont Blanc Pro GTX Goretex Unisex Mountaineering Boots"
              className="-image"
              src="https://www.wildearth.com.au/assets/full/SCA40013.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Scarpa Mont Blanc Pro GTX Goretex Unisex Mountaineering Boots
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                749.95
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/scarpa-terra-gtx-unisex-shoes-brown/SCA00108"
            onClick={[Function]}
          >
            <img
              alt="Scarpa Terra Goretex Unisex Waterproof Hiking Boots - Brown"
              className="-image"
              src="https://www.wildearth.com.au/assets/full/SCA00108.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Scarpa Terra Goretex Unisex Waterproof Hiking Boots - Brown
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                349.95
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/scarpa-delta-leather-fly-mens-hiking-boots-mens-hi/SCA00090"
            onClick={[Function]}
          >
            <img
              alt="Scarpa Delta Leather Fly Mens Waterproof Hiking Boots - T.Di Moro"
              className="-image"
              src="https://www.wildearth.com.au/assets/full/SCA00090.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Scarpa Delta Leather Fly Mens Waterproof Hiking Boots - T.Di Moro
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                399.95
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/scarpa-delta-womens-leather-waterproof-hiking-boot/SCA00065"
            onClick={[Function]}
          >
            <img
              alt="Scarpa Delta Womens Goretex Waterproof Leather Hiking Boots"
              className="-image"
              src="https://www.wildearth.com.au/assets/full/SCA00065.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Scarpa Delta Womens Goretex Waterproof Leather Hiking Boots
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                499.95
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/salomon-quest-prime-gtx-womens-hiking-boots-ebony/411299"
            onClick={[Function]}
          >
            <img
              alt="Salomon Quest Prime GTX Womens Hiking Boots - Ebony/Black/Icy Morn"
              className="-image"
              src="https://www.wildearth.com.au/assets/full/411299.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Salomon Quest Prime GTX Womens Hiking Boots - Ebony/Black/Icy Morn
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                299.99
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/salomon-outblast-ts-cswp-mens-hiking-boots-black-b/409223"
            onClick={[Function]}
          >
            <img
              alt="Salomon OUTblast TS CSWP Mens Hiking Boots - Black/Black/Black"
              className="-image"
              src="https://www.wildearth.com.au/assets/full/409223.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Salomon OUTblast TS CSWP Mens Hiking Boots - Black/Black/Black
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                259.99
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/salomon-quest-prime-gtx-mens-hiking-shoes-phantom/404637"
            onClick={[Function]}
          >
            <img
              alt="Salomon Quest Prime Goretex Mens Waterproof Hiking Boots - Phantom/Black/Quiet Shade"
              className="-image"
              src="https://www.wildearth.com.au/assets/full/404637.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Salomon Quest Prime Goretex Mens Waterproof Hiking Boots - Phantom/Black/Quiet Shade
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                299.99
                 
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="UNX-searchbox__container"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          className="UNX-searchbox__input"
          onChange={[Function]}
          placeholder=""
          value="boots"
        />
        <button
          aria-label="search"
          className="UNX-searchbox__button"
          tabIndex={0}
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
  </div>,
]
`;
